### ISSUE TITLE
Different defaults for SSIMULACRA2 and Butteraugli between APIs and executables

### ISSUE THREAD

**Issue by vrabaud** [libjxl/libjxl#2548]
**Describe the bug**
***SSIMULACRA2***
With images with alpha, `ssimulacra2_main` uses a custom logic https://github.com/libjxl/libjxl/blob/cf760b882e91bbc2f6ab59476e1315fc8db710fb/tools/ssimulacra2_main.cc#L72 while the API  uses a differente one (https://github.com/libjxl/libjxl/blob/cf760b882e91bbc2f6ab59476e1315fc8db710fb/tools/ssimulacra2.cc#L491 value of 0.5 followed by a blend https://github.com/libjxl/libjxl/blob/cf760b882e91bbc2f6ab59476e1315fc8db710fb/tools/ssimulacra2.cc#L432).
What is the correct behavior ? Shouldn't `ssimulacra2_main` just call `ComputeSSIMULACRA2` whatever the input?

***Butteraugli***
`butteraugli_main` uses a default intensity target of `80` (cf https://github.com/libjxl/libjxl/blob/cf760b882e91bbc2f6ab59476e1315fc8db710fb/tools/butteraugli_main.cc#L138) while the default value when creating the official wrapper at https://github.com/libjxl/libjxl/blob/cf760b882e91bbc2f6ab59476e1315fc8db710fb/lib/jxl/butteraugli_wrapper.cc#L72 is `255` (cf https://github.com/libjxl/libjxl/blob/cf760b882e91bbc2f6ab59476e1315fc8db710fb/lib/jxl/common.h#L70).

Also, is there a way to get the Butteraugli distance like done in the internal API at https://github.com/libjxl/libjxl/blob/cf760b882e91bbc2f6ab59476e1315fc8db710fb/lib/jxl/enc_butteraugli_comparator.h#L46 but by using the official API in `include/jxl` ? The closest we have is https://github.com/libjxl/libjxl/blob/cf760b882e91bbc2f6ab59476e1315fc8db710fb/lib/include/jxl/butteraugli.h#L139

**Comment by vrabaud** [comment]
BTW, which implementation of Butteraugli should be used? Isn't https://github.com/google/butteraugli the official one?

**Comment by jonsneyers** [comment]
I like the ssimulacra_main behavior better (try both a dark and a bright background and take worst of two), but it does of course take twice as long. There is no real 'correct' behavior imo.

The butteraugli in the libjxl repo is more recent than the one in the butteraugli repo. I would recommend using the more recent one.

**Comment by vrabaud** [comment]
Thx @jonsneyers . Maybe it would be worth adding a note to the official butteraugli. And if this metric evolves, maybe it could be versioned, just like SSIMULACRA 2 and 2.1

How about the default intensity value in butteraugli? 

Finally, how did you compute this metric for your benchmarks? Through the API or the binary? Which one is the correct way? Thx !

**Comment by jonsneyers** [comment]
I used the binary, but it doesn't matter because all the images were without alpha.

**Comment by fabiorug** [comment]
Probably the image could be 55% less red and 98% more abstract

**Comment by bittorf** [comment]
Is there a way to align both butteraugli values?  
Is seems to be too far away for a drop-in replacement.

```
# from https://github.com/google/butteraugli
$ butteraugli original.png distorted.png 
30.339626

# from https://github.com/libjxl/libjxl/
$ butteraujxl original.png distorted.png 
16.4510307312
3-norm: 6.880581
```

**Comment by jonsneyers** [comment]
No, the butteraugli from github.com/google/butteraugli is effectively just a different metric, it hasn't been updated in 7 years.

The butteraugli from the libjxl repo is more recent and presumably better than the old one, but it's different and it's not just a matter of aligning the scales.
